Remove usage of @AlwaysInline as it's not GraalVM API

### Describe the bug

There are some usages of `@AlwaysInline` that prevent a switch-over of the GraalVM dependency from `org.graalvm.nativeimage:svm` (which isn't supposed to be API) to `org.graalvm.sdk:graal-sdk`. These are currently:

```
grep -rn @AlwaysInline
core/runtime/src/main/java/io/quarkus/runtime/graal/ConfigurationSubstitutions.java:26:    @AlwaysInline("trivial")
extensions/jdbc/jdbc-mssql/runtime/src/main/java/io/quarkus/jdbc/mssql/runtime/graal/com/microsoft/sqlserver/jdbc/SQLServerJDBCSubstitutions.java:90:    @AlwaysInline("We need this to be constant folded")
```

### Expected behavior

No more use of `@AlwaysInline` and `graal-sdk` is being used as a dep only.

### Actual behavior

Some modules fail to compile with the `graal-sdk` dependency. For example:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/extensions/jdbc/jdbc-mssql/runtime/src/main/java/io/quarkus/jdbc/mssql/runtime/graal/com/microsoft/sqlserver/jdbc/SQLServerJDBCSubstitutions.java:[12,27] cannot find symbol
  symbol:   class AlwaysInline
  location: package com.oracle.svm.core
[ERROR] /home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/extensions/jdbc/jdbc-mssql/runtime/src/main/java/io/quarkus/jdbc/mssql/runtime/graal/com/microsoft/sqlserver/jdbc/SQLServerJDBCSubstitutions.java:[90,6] cannot find symbol
  symbol:   class AlwaysInline
  location: class io.quarkus.jdbc.mssql.runtime.graal.com.microsoft.sqlserver.jdbc.DisableFMTRemove
```

### How to Reproduce?

Change the deps to `graal-sdk` over `svm` artifact. Compile quarkus.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

22.3.2

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

See https://github.com/oracle/graal/pull/4683 which moved supported annotations to the `graal-sdk`.

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/ConfigurationSubstitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/ConfigurationSubstitutions.java
@@ -2,7 +2,6 @@ package io.quarkus.runtime.graal;
 
 import org.eclipse.microprofile.config.Config;
 
-import com.oracle.svm.core.AlwaysInline;
 import com.oracle.svm.core.annotate.Alias;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
@@ -23,7 +22,6 @@ final class Target_io_smallrye_config_SmallRyeConfigProviderResolver {
     }
 
     @Substitute
-    @AlwaysInline("trivial")
     public Config getConfig(ClassLoader classLoader) {
         return getConfig();
     }

--- a/extensions/jdbc/jdbc-mssql/runtime/src/main/java/io/quarkus/jdbc/mssql/runtime/graal/com/microsoft/sqlserver/jdbc/SQLServerJDBCSubstitutions.java
+++ b/extensions/jdbc/jdbc-mssql/runtime/src/main/java/io/quarkus/jdbc/mssql/runtime/graal/com/microsoft/sqlserver/jdbc/SQLServerJDBCSubstitutions.java
@@ -9,7 +9,6 @@ import javax.net.ssl.KeyManager;
 import com.microsoft.sqlserver.jdbc.SQLServerException;
 import com.microsoft.sqlserver.jdbc.SQLServerStatement;
 import com.microsoft.sqlserver.jdbc.SqlAuthenticationToken;
-import com.oracle.svm.core.AlwaysInline;
 import com.oracle.svm.core.annotate.Delete;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
@@ -87,7 +86,6 @@ final class SQLServerFMTQuery {
 final class DisableFMTRemove {
 
     @Substitute
-    @AlwaysInline("We need this to be constant folded")
     public final boolean getUseFmtOnly() throws SQLServerException {
         return false;//Important for this to be disabled via a constant
     }

